Index: librdkafka-1.1.0/src/rdkafka_msgset_writer.c
===================================================================
--- librdkafka-1.1.0.orig/src/rdkafka_msgset_writer.c
+++ librdkafka-1.1.0/src/rdkafka_msgset_writer.c
@@ -107,7 +107,7 @@ rd_kafka_msgset_writer_select_MsgVersion
                 int feature;
                 int16_t ApiVersion;
         } compr_req[RD_KAFKA_COMPRESSION_NUM] = {
-                [RD_KAFKA_COMPRESSION_LZ4] = { RD_KAFKA_FEATURE_LZ4, 3 },
+                [RD_KAFKA_COMPRESSION_LZ4] = { RD_KAFKA_FEATURE_LZ4, 2 },
 #if WITH_ZSTD
                 [RD_KAFKA_COMPRESSION_ZSTD] = { RD_KAFKA_FEATURE_ZSTD, 7 },
 #endif
