--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -524,7 +524,7 @@ option(WITH_MGR "ceph-mgr is enabled" ON)
 if(WITH_MGR)
   # Please specify 3 or 3.[0-7] if you want to build with python3 support.
   # FindPyhonInterp and FindPythonLibs think they belong to different families.
-  set(MGR_PYTHON_VERSION "2.7" CACHE
+  set(MGR_PYTHON_VERSION "3.8" CACHE
     STRING "minimal required version of python runtime for running mgr plugins. ")
   find_package(PythonInterp ${MGR_PYTHON_VERSION} REQUIRED)
   find_package(PythonLibs ${MGR_PYTHON_VERSION} REQUIRED)
