--- a/src/global/signal_handler.h	2021-09-22 12:44:38.418821002 +0000
+++ b/src/global/signal_handler.h	2021-09-22 12:46:52.295182576 +0000
@@ -16,15 +16,12 @@
 #define CEPH_GLOBAL_SIGNAL_HANDLER_H
 
 #include <signal.h>
+#include <string.h>
 #include "acconfig.h"
 
 typedef void (*signal_handler_t)(int);
 
-#ifndef HAVE_REENTRANT_STRSIGNAL
-# define sig_str(signum) sys_siglist[signum]
-#else
-# define sig_str(signum) strsignal(signum)
-#endif
+#define sig_str(signum) strsignal(signum)
 
 void install_sighandler(int signum, signal_handler_t handler, int flags);
 
