diff -Naur ceph-13.2.10/src/mgr/PythonCompat.h ceph-13.2.10-r1/src/mgr/PythonCompat.h
--- ceph-13.2.10/src/mgr/PythonCompat.h 2021-10-11 14:01:49.437453665 +0000
+++ ceph-13.2.10-r1/src/mgr/PythonCompat.h      2021-10-11 14:03:21.951562008 +0000
@@ -13,7 +13,7 @@
 inline PyObject* PyString_FromString(const char *v) {
   return PyUnicode_FromFormat("%s", v);
 }
-inline char* PyString_AsString(PyObject *string) {
+inline const char* PyString_AsString(PyObject *string) {
   return (char *)PyUnicode_AsUTF8(string);
 }
 inline long PyInt_AsLong(PyObject *io) {
