diff -Naur portage-3.0.30/lib/_emerge/actions.py portage-3.0.30-r1/lib/_emerge/actions.py
--- portage-3.0.30/lib/_emerge/actions.py	2021-12-12 08:05:48.000000000 +0000
+++ portage-3.0.30-r1/lib/_emerge/actions.py	2022-01-04 13:36:14.129977736 +0000
@@ -3478,7 +3478,6 @@
         spinner.update = spinner.update_scroll
 
     if "--quiet" not in emerge_config.opts:
-        portage.deprecated_profile_check(settings=emerge_config.target_config.settings)
         repo_name_check(emerge_config.trees)
         repo_name_duplicate_check(emerge_config.trees)
         config_protect_check(emerge_config.trees)
