diff --git a/internal/types/host/keepalived_host_collector_host.go b/internal/types/host/keepalived_host_collector_host.go
index ea66e45..8f48f36 100644
--- a/internal/types/host/keepalived_host_collector_host.go
+++ b/internal/types/host/keepalived_host_collector_host.go
@@ -45,11 +45,11 @@ func NewKeepalivedHostCollectorHost(useJSON bool, pidPath string) *KeepalivedHos

 func (k *KeepalivedHostCollectorHost) Refresh() error {
 	if k.useJSON {
-		if err := k.signal(k.SIGJSON); err != nil {
-			logrus.WithError(err).Error("Failed to send JSON signal to keepalived")
+		//if err := k.signal(k.SIGJSON); err != nil {
+		//	logrus.WithError(err).Error("Failed to send JSON signal to keepalived")

-			return err
-		}
+		//	return err
+		//}

 		return nil
 	}
@@ -57,13 +57,13 @@ func (k *KeepalivedHostCollectorHost) Refresh() error {
 	if err := k.signal(k.SIGSTATS); err != nil {
 		logrus.WithError(err).Error("Failed to send STATS signal to keepalived")

-		return err
+		// return err
 	}

 	if err := k.signal(k.SIGDATA); err != nil {
 		logrus.WithError(err).Error("Failed to send DATA signal to keepalived")

-		return err
+		//return err
 	}

 	return nil
@@ -157,7 +157,7 @@ func (k *KeepalivedHostCollectorHost) JSONVrrps() ([]collector.VRRP, error) {

 	f, err := os.Open(fileName)
 	if err != nil {
-		logrus.WithError(err).WithField("fileName", fileName).Error("failed to open JSON VRRP file")
+		//logrus.WithError(err).WithField("fileName", fileName).Error("failed to open JSON VRRP file")

 		return nil, err
 	}