--- a/3rdparty/grpc-1.10.0/src/core/lib/gpr/log_linux.cc	2020-10-29 14:17:18.844015734 -0000
+++ b/3rdparty/grpc-1.10.0/src/core/lib/gpr/log_linux.cc	2020-10-29 14:17:23.204015785 -0000
@@ -39,8 +39,6 @@
 #include <time.h>
 #include <unistd.h>
 
-static long gettid(void) { return syscall(__NR_gettid); }
-
 void gpr_log(const char* file, int line, gpr_log_severity severity,
              const char* format, ...) {
   char* message = nullptr;
