Index: R-3.4.1/m4/cairo.m4
===================================================================
--- R-3.4.1.orig/m4/cairo.m4
+++ R-3.4.1/m4/cairo.m4
@@ -70,7 +70,6 @@ else
                     [AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
-#include <cairo-xlib.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
@@ -143,7 +142,6 @@ int main(void) {
 		     [r_cv_cairo_works], 
                      [AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <cairo.h>
-#include <cairo-xlib.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
Index: R-3.4.1/src/modules/X11/devX11.h
===================================================================
--- R-3.4.1.orig/src/modules/X11/devX11.h
+++ R-3.4.1/src/modules/X11/devX11.h
@@ -61,7 +61,9 @@ typedef enum {
 # else
 #  include <cairo.h>
 # endif
+# ifdef HAVE_X11
 #  include <cairo-xlib.h>
+# endif
 #endif
 
 Rboolean X11DeviceDriver(pDevDesc, const char*, double, double, double,
