--- a/src/crc32.c
+++ b/src/crc32.c
@@ -19,7 +19,7 @@
       errmsg("crc32 value cannot exceed 32 bits")));
   }

-  crc32 *out = palloc(sizeof(crc32));
+  uint32 out;


   char *p;
@@ -35,7 +35,7 @@
   /* I don't check if pout overflows uint32 because it's restricted by string
    * length check above.
    */
-  out = (uint32_t)pout;
+  out = (uint32) pout;

   PG_RETURN_CRC32(out);
 }
@@ -44,7 +44,7 @@
 Datum
 crc32_out(PG_FUNCTION_ARGS)
 {
-  crc32 *in = PG_GETARG_CRC32(0);
+  crc32 in = PG_GETARG_CRC32(0);
   char *out = (char *) palloc(10);
   snprintf(out, 10, "%08x", in);
   PG_RETURN_CSTRING(out);
--- a/src/common.c
+++ b/src/common.c
@@ -7,6 +7,9 @@
  * Portions Copyright (c) 2001-2008, PostgreSQL Global Development Group
  */
 #include "postgres.h"
+#if PG_VERSION_NUM >= 160000
+#include "varatt.h"
+#endif

 #include "common.h"
 #include "fmgr.h"
--- a/src/md5.c
+++ b/src/md5.c
@@ -7,6 +7,9 @@
  * Portions Copyright (c) 2001-2009, PostgreSQL Global Development Group
  */
 #include "postgres.h"
+#if PG_VERSION_NUM >= 160000
+#include "varatt.h"
+#endif

 #include "access/hash.h"
 #include "common.h"
--- a/src/sha.c
+++ b/src/sha.c
@@ -12,6 +12,9 @@
  * Portions Copyright (c) 2001-2009, PostgreSQL Global Development Group
  */
 #include "postgres.h"
+#if PG_VERSION_NUM >= 160000
+#include "varatt.h"
+#endif

 #include "access/hash.h"
 #include "common.h"
