diff --git a/Makefile b/Makefile
index 31eae06..c4bde74 100644
--- a/Makefile
+++ b/Makefile
@@ -49,7 +49,7 @@ else
 	endif
 endif
 
-PROMU := $(FIRST_GOPATH)/bin/promu --config $(PROMU_CONF)
+PROMU := `which promu` --config $(PROMU_CONF)
 
 e2e-out = collector/fixtures/e2e-output.txt
 ifeq ($(MACH), ppc64le)
diff --git a/Makefile.common b/Makefile.common
index db98993..dba8134 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -59,7 +59,7 @@ $(warning Some recipes may not work as expected as the current Go runtime is '$(
 		GOVENDOR := $(FIRST_GOPATH)/bin/govendor
 	endif
 endif
-PROMU        := $(FIRST_GOPATH)/bin/promu
+PROMU        := `which promu`
 pkgs          = ./...
 
 ifeq (arm, $(GOHOSTARCH))
@@ -199,12 +199,12 @@ endif
 endif
 
 .PHONY: common-build
-common-build: promu
+common-build:
 	@echo ">> building binaries"
 	GO111MODULE=$(GO111MODULE) $(PROMU) build --prefix $(PREFIX)
 
 .PHONY: common-tarball
-common-tarball: promu
+common-tarball:
 	@echo ">> building release tarball"
 	$(PROMU) tarball --prefix $(PREFIX) $(BIN_DIR)
 
